src/Storages/MergeTree/ReplicatedMergeTreeTableMetadata.cpp

```diff
@@ -456,6 +456,14 @@ StorageInMemoryMetadata ReplicatedMergeTreeTableMetadata::Diff::getNewMetadata(c
         new_metadata.table_ttl = TTLTableDescription::getTTLForTableFromAST(
             new_metadata.table_ttl.definition_ast, new_metadata.columns, context, new_metadata.primary_key);
 
+    if (!projections_changed)
```

Comment:
No, because we mutate the Metadata and use it instantly for alter. For replicated, we mutate the metadata, create the string for ZK, and then recreate metadata from that string. The bug was in the recreate part.